core-web/apps/dotcms-ui/src/app/portlets/dot-containers/dot-container-create/dot-container-properties/dot-container-properties.component.ts

```diff
@@ -64,9 +65,26 @@ export class DotContainerPropertiesComponent implements OnInit {
                 }
             });
 
-        this.form.valueChanges.pipe(takeUntil(this.destroy$)).subscribe((values) => {
-            this.store.updateFormStatus({ invalidForm: !this.form.valid, container: values });
-        });
+        //
+        this.form.valueChanges
+            .pipe(takeUntil(this.destroy$), startWith(this.form.value), pairwise())
+            .subscribe(([prevValue, currValue]) => {
+                this.store.updateFormStatus({
+                    invalidForm: !this.form.valid,
+                    container: currValue
+                });
+                if (
+                    (prevValue.maxContentlets === 0 || prevValue.maxContentlets === null) &&
+                    currValue.maxContentlets > 0
```

Comment:
We can encapsulate complex conditions for cleaner code: https://github.com/ryanmcdermott/clean-code-javascript#encapsulate-conditionals

Reply:
Noted

Reply:
updated